backend/api/projects.py

```diff
@@ -46,27 +35,40 @@ def add_projects():
                     manifestlist.append(json.loads(file.read().decode('utf-8')))
                 elif file.filename.endswith('.json5'):
                     manifestlist.append(json5.loads(file.read().decode('utf-8')))
-            return_ids = uploader.save_manifest_to_db(manifestlist)
-            g.rerun_saved_searches()
-            return jsonify(return_ids)
+            projects = uploader.save_manifest_to_db(manifestlist)
+            for project in projects:
+                if 'comments' not in project:
+                    project['comments'] = []
+                g.projects.insert(project)
+                add_notification(current_user['email'], project['authors'],
+                                 project['_id'], "create", reason='author')
+                add_self_action(current_user['email'], project['_id'], "create")
+                g.rerun_saved_searches(current_user['email'], project['_id'], "create")
+            return jsonify([project['_id'] for project in projects])
         except Exception as err:
             raise ApiException(str(err), 400)
 
     else:
         try:
-            return_ids = []
+            projects = []
             if ('application/json' in request.content_type) and\
                     ('application/json5' not in request.content_type):
-                return_ids = uploader.save_manifest_to_db(request.get_json())
+                projects = uploader.save_manifest_to_db(request.get_json())
             elif 'application/json5' in request.content_type:
-                return_ids = uploader.save_manifest_to_db(
+                projects = uploader.save_manifest_to_db(
                     json5.loads(request.data.decode('utf-8')))
             else:
                 raise ApiException("Wrong content header" +
                                    "and no files attached", 400)
-
-            g.rerun_saved_searches()
-            return jsonify(return_ids)
+            project = projects[0]
+            if 'comments' not in project:
+                project['comments'] = []
+            g.projects.insert(project)
```

Comment:
Fixed..... Was trappend in a different understanding of code.... In that actual case we have unreachable code which was planned to work dirrently...